File: backend/components/metiers/administration_publique/utils/metrics/samples/sample_usage.js
// sample_usage.js – Exemples ultra avancés pour le module metrics (clé en main)
const metrics = require('../core/metrics');
const data = require('./sample_metrics_data.json').metrics;

// Calcul de la moyenne
console.log('Moyenne:', metrics.calculerMoyenne(data));

// Calcul de la médiane
console.log('Médiane:', metrics.calculerMediane(data));

// Gestion d’erreur (données vides)
try {
  metrics.calculerMoyenne([]);
} catch (e) {
  console.error('Erreur attendue:', e.message);
}

File: backend/components/metiers/threed/utils/metrics/samples/sample_usage.py
# sample_usage.py – Exemples ultra avancés pour le module metrics (clé en main)
from ..core import metrics
import json
with open('sample_metrics_data.json') as f:
    data = json.load(f)["metrics"]

print('Moyenne:', metrics.calculer_moyenne(data))
print('Médiane:', metrics.calculer_mediane(data))

try:
    metrics.calculer_moyenne([])
except Exception as e:
    print('Erreur attendue:', e)
